### PR TITLE
#21 testing user routes

### PR DESCRIPTION
**Pull request by roboMoose**
Routes should be solid (tested with postman). Still trying to get tests to behave.

### CODE REVIEW

File: server/db/index.js [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-const db = require('./db');
+const db = require("./db");
 //MODELS
-const Cart = require('./models/Cart');
-const Guest = require('./models/Guest');
-const Product = require('./models/Product');
-const Review = require('./models/Review');
-const Transaction = require('./models/Transaction');
-const User = require('./models/User');
+const Cart = require("./models/Cart");
```
Comment (nicknapoli82) [inline server/db/index.js]: Though I have uncertainty at the moment. I feel confident we don't want to comment out these associations

File: server/server.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 const express = require("express");
 const app = express();
+
 const path = require("path");
-const PORT = process.env.PORT || 3000;
+const port = process.env.PORT || 3000;
+app.listen(port, () => console.log(`Dragons spotted over on port ${port}`));
```
Comment (wallacepreston) [inline server/server.js]: Are you certain you want to move the app.listen to this line?  I believe any middleware/routes after this line won't be included in your app.  Make sure to test your api routes to make sure this works.

File: server/tests/userRoute.spec.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+describe("GET Users", () => {
+  it("returns a full list of current members", async () => {
+    const response = await request(app).get("/api/users");
+    expect(response.status).toEqual(200);
+    expect(response.body.length).toBeGreaterThan(1);
+  });
+});
+
+describe("POST new user", () => {
```
Comment (wallacepreston) [inline server/tests/userRoute.spec.js]: Good tests!

File: server/server.js (hunk 2 above)
Comment (roboMoose) [reply server/server.js]: There is an issue we are having with jest, and it really doesn't like app.listen for some reason. I would normally have it at the bottom of the page.

File: server/routes/users.js [hunk 4]
```diff
@@ -49,7 +48,7 @@ router.delete("/:id", async (req, res, next) => {
         id: req.params.id
       }
     });
-    res.sendStatus(204);
+    res.status(404).send("user removed");
```
Comment (nicknapoli82) [inline server/routes/users.js]: I'm not sure why we are sending 404 here, but I'll roll with it for now.

**Pull request merged**